org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/text/correction/QuickAssistProcessor.java

```diff
@@ -981,252 +978,19 @@ public static boolean getRemoveVarOrInferredLambdaParameterTypes(IInvocationCont
-		// ImportRewrite importRewrite= proposal.createImportRewrite(context.getASTRoot());
-		ImportRewrite importRewrite= StubUtility.createImportRewrite(astRoot, true);
-
-		VariableDeclarationStatement variableDeclarationStatement= ASTNodes.getParent(methodReferenceNode, VariableDeclarationStatement.class);
-		MethodInvocation methodInvocationNode= ASTNodes.getParent(methodReferenceNode, MethodInvocation.class);
-		Assignment variableAssignment= ASTNodes.getParent(methodReferenceNode, Assignment.class);
-
-		if ((variableAssignment != null || variableDeclarationStatement != null) && methodInvocationNode == null) {
-			/*
-			 * variable declaration
-			 */
-			Type type= null;
-			ReturnType returnType= null;
-			if (variableDeclarationStatement != null) {
-				type= variableDeclarationStatement.getType();
-				returnType= getReturnType(ast, importRewrite, type);
-			} else {
-				Expression leftHandSide= variableAssignment.getLeftHandSide();
-				ITypeBinding assignmentTypeBinding= leftHandSide.resolveTypeBinding();
-				if (assignmentTypeBinding == null) {
-					return false;
-				}
-				type= importRewrite.addImport(assignmentTypeBinding, ast);
-				returnType= new ReturnType();
-				returnType.type= type;
-				returnType.binding= assignmentTypeBinding;
-			}
-			if (returnType.binding == null) {
-				return false;
-			}
-			MethodDeclaration newMethodDeclaration= ast.newMethodDeclaration();
-			newMethodDeclaration.setName((SimpleName) rewrite.createCopyTarget(methodReferenceNode.getName()));
-			newMethodDeclaration.modifiers().add(ast.newModifier(ModifierKeyword.PRIVATE_KEYWORD));
-			if (addStaticModifier) {
-				newMethodDeclaration.modifiers().add(ast.newModifier(ModifierKeyword.STATIC_KEYWORD));
-			}
-
-			IMethodBinding functionalInterfaceMethod= variableDeclarationStatement == null
-					? returnType.binding.getFunctionalInterfaceMethod()
-					: variableDeclarationStatement.getType().resolveBinding().getFunctionalInterfaceMethod();
-			if (functionalInterfaceMethod != null) {
-				returnType.type= importRewrite.addImport(functionalInterfaceMethod.getReturnType(), ast);
-				returnType.binding= functionalInterfaceMethod.getReturnType();
-				ITypeBinding[] typeArguments= functionalInterfaceMethod.getParameterTypes();
-				for (int i= 0; i < typeArguments.length; i++) {
-					ITypeBinding iTypeBinding= typeArguments[i];
-					SingleVariableDeclaration newSingleVariableDeclaration= ast.newSingleVariableDeclaration();
-					newSingleVariableDeclaration.setName(ast.newSimpleName(iTypeBinding.getErasure().getName().toLowerCase() + (i + 1)));
-					newSingleVariableDeclaration.setType(importRewrite.addImport(iTypeBinding.getErasure(), ast));
-					newMethodDeclaration.parameters().add(newSingleVariableDeclaration);
-				}
+		AddMissingMethodDeclarationFixCore fix= AddMissingMethodDeclarationFixCore.createAddMissingMethodDeclaration(context.getASTRoot(), covering);
+		if (fix != null) {
+			if (resultingCollections == null) {
```

Comment:
@jjohnstn looking at the code `createAddMissingMethodDeclaration` I feel that the overhead of creating the instance `AddMissingMethodDeclarationFixCore` is less, therefore I used the instance nullability to check if the fix is valid or not. Let me know if you think otherwise

Reply:
That is fine.  Thanks.